frame/support/src/traits.rs

```diff
@@ -331,10 +336,71 @@ pub trait Imbalance<Balance>: Sized + TryDrop {
 	/// is guaranteed to be at most `amount` and the second will be the remainder.
 	fn split(self, amount: Balance) -> (Self, Self);
 
+	/// Consume `self` and return two independent instances; the amounts returned will be in
+	/// approximately the same ratio as `first`:`second`.
+	///
+	/// NOTE: This requires up to `first + second` room for a multiply. Overflow will safely
+	/// saturate.
+	fn ration(self, first: u32, second: u32) -> (Self, Self)
```

Comment:
if wanted we could implement it without saturation similarly to how perbill implements mul.

Reply:
not immediately needed, any it would slow things down, and at least the polkadot and kusama balance types are designed specifically so that there is a u32's worth of space at the high end, so i'd prefer not to at this stage.